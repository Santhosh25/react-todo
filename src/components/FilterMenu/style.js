import styled from "styled-components";

export const Filter = styled.div`
  display: flex;
  justify-content: flex-end;
  color: #fff;
  & span {
    cursor: pointer;
  }
`;
export const FilterLabel = styled.span`
  display: inline-block;
  margin-right: 10px;
`;