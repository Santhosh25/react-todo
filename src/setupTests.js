// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, shallow, mount, render } from "enzyme";
import "jest-styled-components";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { store } from "./store";

React.useLayoutEffect = React.useEffect;
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useLayoutEffect: jest.requireActual("react").useEffect,
}));
// const constantDate = new Date('2017-06-13T04:41:20');
// global.Date = class extends Date {
//   constructor(value) {
//     return constantDate;
//   }
// };
configure({ adapter: new Adapter() });

const wrapChildWithThemeProvider = (children) => (
  <Provider store={store}>{children}</Provider>
);

const shallowWithTheme = (children) =>
  shallow(wrapChildWithThemeProvider(children));
const mountWithTheme = (children) =>
  mount(wrapChildWithThemeProvider(children));
const renderWithTheme = (children) =>
  render(wrapChildWithThemeProvider(children));

export { shallowWithTheme, mountWithTheme, renderWithTheme };
