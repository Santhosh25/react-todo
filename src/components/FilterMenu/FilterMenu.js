import React, { useEffect, useState } from "react";
import Icon from "@material-ui/core/Icon";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import Tooltip from "@material-ui/core/Tooltip";
import { useDispatch, useSelector } from "react-redux";

import actions from "../../actions";
import { Filter, FilterLabel } from "./style";

const options = ["All", "Pending", "Completed", "Archived"];
const action = {
  All: "FILTER_ALL",
  Pending: "FILTER_PENDING",
  Completed: "FILTER_COMPLETED",
  Archived: "FILTER_ARCHIVED",
};

const ITEM_HEIGHT = 48;

const FilterMenu = () => {
  const todos = useSelector((state) => state.todos.data);
  const actionType = useSelector((state) => state.filters.actionType);
  const [anchorEl, setAnchorEl] = useState(null);
  const [filterValue, setFilterValue] = useState("All");
  const open = Boolean(anchorEl);
  const dispatch = useDispatch();

  const handleClick = (event) => {
    setAnchorEl(event.currentTarget);
  };

  const handleClose = (type) => {
    setAnchorEl(null);
    if (type) {
      setFilterValue(type);
      dispatch({
        type: action[type],
        payload: todos,
      });
    }
  };

  useEffect(() => {
    dispatch({
      type: actions.FILTER_ALL,
      payload: todos,
    });
    setFilterValue("All");
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [todos, dispatch]);

  useEffect(() => {
    if (actionType === actions.SEARCH) {
      setFilterValue("All");
    }
  }, [actionType]);

  return (
    <Filter>
      <FilterLabel>{filterValue}</FilterLabel>
      <Tooltip title="Filter List">
        <Icon
          aria-label="more"
          aria-controls="long-menu"
          aria-haspopup="true"
          onClick={handleClick}
        >
          filter_list
        </Icon>
      </Tooltip>
      <Menu
        id="long-menu"
        anchorEl={anchorEl}
        keepMounted
        open={open}
        onClose={(e) => handleClose("")}
        PaperProps={{
          style: {
            maxHeight: ITEM_HEIGHT * 4.5,
            width: "20ch",
          },
        }}
      >
        {options.map((option) => (
          <MenuItem
            key={option}
            selected={option === filterValue}
            onClick={(e) => handleClose(option)}
          >
            {option}
          </MenuItem>
        ))}
      </Menu>
    </Filter>
  );
};

export default React.memo(FilterMenu);
