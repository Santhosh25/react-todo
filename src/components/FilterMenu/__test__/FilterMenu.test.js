import React from "react";
import FilterMenu from "../FilterMenu";
import { renderWithTheme } from "../../../setupTests";

describe("FilterMenu Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = renderWithTheme(<FilterMenu />);
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
