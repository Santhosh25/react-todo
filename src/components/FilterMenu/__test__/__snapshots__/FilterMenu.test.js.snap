// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterMenu Component renders correctly 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "sc-bdnxRM dUFXLy",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "sc-gtsrHT bucyKh",
        },
        "children": Array [
          Node {
            "data": "Pending",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "aria-controls": "long-menu",
            "aria-haspopup": "true",
            "aria-hidden": "true",
            "aria-label": "more",
            "class": "material-icons MuiIcon-root",
            "title": "Filter List",
          },
          "children": Array [
            Node {
              "data": "filter_list",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-controls": undefined,
            "aria-haspopup": undefined,
            "aria-hidden": undefined,
            "aria-label": undefined,
            "class": undefined,
            "title": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-controls": undefined,
            "aria-haspopup": undefined,
            "aria-hidden": undefined,
            "aria-label": undefined,
            "class": undefined,
            "title": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "aria-controls": "long-menu",
          "aria-haspopup": "true",
          "aria-hidden": "true",
          "aria-label": "more",
          "class": "material-icons MuiIcon-root",
          "title": "Filter List",
        },
        "children": Array [
          Node {
            "data": "filter_list",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "sc-gtsrHT bucyKh",
          },
          "children": Array [
            Node {
              "data": "Pending",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "text",
            },
          ],
          "name": "span",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-controls": undefined,
          "aria-haspopup": undefined,
          "aria-hidden": undefined,
          "aria-label": undefined,
          "class": undefined,
          "title": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-controls": undefined,
          "aria-haspopup": undefined,
          "aria-hidden": undefined,
          "aria-label": undefined,
          "class": undefined,
          "title": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
